### PR TITLE
Refactor/data/validation report

### PR DESCRIPTION
**Pull request by ramirobg94**

### CODE REVIEW

File: webapp/store/survey/state.js [hunk 1]
```diff
@@ -20,6 +20,8 @@ export const getSurveyDefaultLang = R.pipe(getSurveyInfo, Survey.getDefaultLangu
 
 export const getSurveyId = R.pipe(getSurvey, Survey.getId)
 
+export const getNodeDefs = R.pipe(getSurvey, Survey.getNodeDefs)
```
Comment (minotogna) [inline webapp/store/survey/state.js:23]: you don't need to add this. You can just use Survey.getNodeDefs

File: webapp/views/App/views/Data/Records/Records.js [hunk 2]
```diff
@@ -1,26 +1,26 @@
...
+import HeaderLeft from './HeaderLeft'
+import RowHeader from './RowHeader'
+import Row from './Row'
 
-const RecordsView = () => {
+const Records = () => {
   const history = useHistory()
   const cycle = useSurveyCycleKey()
-  const nodeDefKeys = [] // TODO: fetch them
+  const nodeDefKeys = useSurveyNodeDefs()
```
Comment (minotogna) [inline webapp/views/App/views/Data/Records/Records.js:21]: add a hook useNodeDefRootKeys that returns an array of nodeDefs that are keys of the root entity

File: webapp/store/survey/hooks/index.js [hunk 3]
```diff
@@ -19,6 +19,7 @@ export const useOnSurveyCycleUpdate = (effect) => {
   const surveyCycleKey = useSurveyCycleKey()
   useOnUpdate(effect, [surveyCycleKey])
 }
+export const useSurveyNodeDefs = () => useSelector(SurveyState.getNodeDefs)
```
Comment (minotogna) [inline webapp/store/survey/hooks/index.js:22]: this is not needed

File: webapp/views/App/views/Data/Records/Row/Row.js [hunk 4]
```diff
@@ -13,12 +13,12 @@ import * as Record from '@core/record/record'
...
 import * as DateUtils from '@core/dateUtils'
-import { useSurveyInfo } from '@webapp/store/survey'
+import { useSurveyInfo, useSurveyNodeDefs } from '@webapp/store/survey'
 import { useUser } from '@webapp/store/user'
 
-const RecordsRow = (props) => {
+const Row = (props) => {
   const { row: record, rowNo } = props
-  const nodeDefKeys = [] // TODO
+  const nodeDefs = useSurveyNodeDefs()
```
Comment (minotogna) [inline webapp/views/App/views/Data/Records/Row/Row.js:21]: same as Records

File: webapp/views/App/views/Data/Records/Records.js (hunk 2 above)
Comment (ramirobg94) [reply webapp/views/App/views/Data/Records/Records.js:21]: if i understand well this are the keys of the root entity:

Captura de pantalla 2020-06-17 a las 9 47 01

Comment (minotogna) [reply webapp/views/App/views/Data/Records/Records.js]: Nope. By keys it’s meant the attribute that are marked as key

File: webapp/views/App/views/Data/Records/Records.js [hunk 5]
```diff
@@ -1,26 +1,26 @@
...
 
-const RecordsView = () => {
+const Records = () => {
   const history = useHistory()
   const cycle = useSurveyCycleKey()
-  const nodeDefKeys = [] // TODO: fetch them
+  const nodeDefKeys = useNodeDefRootKeys()
 
-  const noCols = 3 + nodeDefKeys.length
+  const noCols = 3 + Object.keys(nodeDefKeys).length
```
Comment (minotogna) [inline webapp/views/App/views/Data/Records/Records.js:23]: nodeDefKeys should be an array already. Object.keys might not be needed

File: webapp/views/App/views/Data/Records/Row/Row.js [hunk 6]
```diff
@@ -31,7 +31,7 @@ const RecordsRow = (props) => {
         <ErrorBadge validation={Validation.getValidation(record)} showLabel={false} className="error-badge-inverse" />
         {rowNo}
       </div>
-      {nodeDefKeys.map((nodeDef) => (
+      {Object.values(nodeDefs).map((nodeDef) => (
```
Comment (minotogna) [inline webapp/views/App/views/Data/Records/Row/Row.js:34]: nodeDefKeys should be an array already. Object.values might not be needed

File: webapp/views/App/views/Data/Records/RowHeader/RowHeader.js [hunk 7]
```diff
@@ -3,19 +3,18 @@ import React from 'react'
...
   const lang = useLang()
   return (
     <>
       <div>#</div>
-      {
-        nodeDefKeys.map((k) => (
-          <div key={NodeDef.getUuid(k)}>{NodeDef.getLabel(k, lang)}</div>
-        )) // TODO use SurveyState.getNodeDefLabel
-      }
+      {Object.values(nodeDefs).map((nodeDef) => (
```
Comment (minotogna) [inline webapp/views/App/views/Data/Records/RowHeader/RowHeader.js:15]: nodeDefKeys should be an array already. Object.values might not be needed

File: webapp/store/survey/hooks/index.js [hunk 8]
```diff
@@ -20,6 +20,8 @@ export const useOnSurveyCycleUpdate = (effect) => {
   useOnUpdate(effect, [surveyCycleKey])
 }
 
+export const useNodeDefRootKeys = () => Survey.getNodeDefKeys(Survey.getNodeDefRoot(useSurvey()))(useSurvey())
```
Comment (minotogna) [inline webapp/store/survey/hooks/index.js:23]: this must be
```export const useNodeDefRootKeys = () => Survey.getNodeDefRootKeys(useSurvey())```

**Pull request merged**